skiko/src/jvmMain/cpp/linux/SoftwareRedrawer.cc

```diff
@@ -0,0 +1,89 @@
+#include <jawt_md.h>
+#include <X11/Xresource.h>
+#include <stdint.h>
+#include "jni_helpers.h"
+
+#include "SkSurface.h"
+#include "src/core/SkAutoMalloc.h"
+
+class SoftwareDevice
+{
+public:
+    Display* display;
+    Window window;
+    GC gc;
+    SkAutoMalloc surfaceMemory;
+    sk_sp<SkSurface> surface;
+
+    ~SoftwareDevice() {
+        surface.reset(nullptr);
```

Comment:
Must call https://tronche.com/gui/x/xlib/GC/XFreeGC.html I guess, and surface.reset() isn't needed.

Reply:
I'm not sure about `XFreeGC`, because AWT should manage the removal of` display` and `gc`. We don't create them, we just get them from AWT.

Reply:
Hm `device->gc = XCreateGC(device->display, device->window, 0, nullptr);`?

Reply:
Done.